### PR TITLE
Truncate long plan labels and refer to "print-plans"

### LINKED ISSUE #2710: [BUG] dot file generation can go over the limits of dot
**Describe the bug**
I recently hit a bug when trying to build a dot file generated by the tool for NDS query 24.

```
$ dot -Tsvg -O ./profile_output_171/*.dot
Error: ./profile_output_171/app-20210528152635-0280-query-24.dot: syntax error in line 3 scanning a quoted string (missing endquote? longer than 16384?)
String starting:"\lApplication: app-20210528152635-0280\lQuery: 24\l\l== Parsed Logical Plan ==\
```

I had to go in and delete the extra info from the label for the digraph.

It might be nice to detect this and avoid it, or even better come up with a different way to display the plans that would not violate this limit.

### PR DESCRIPTION
**Pull request by gerashegalov**
Properly treat graphviz compiled with the default 16K label max

Closes #2710.

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateDot.scala [hunk 1]
```diff
@@ -292,6 +284,39 @@ object SparkPlanGraph {
...
   }
+
+  def createDotLabel(
+    outputDir: String, 
+    appId: String, 
+    sqlId: String,
+    physicalPlan: String,
+    maxLength: Int = 16384 
+  ): String = {
+    val sqlPlanFile = s"file://$outputDir/planDescriptions-$appId"
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateDot.scala:295]: outputDir isn't necessarily file://, could be in hdfs for instance, I guess maybe you are thinking they download it anyway so when open it should be?  But if they do that outputdir could be different

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateDot.scala [hunk 2]
```diff
@@ -292,6 +284,39 @@ object SparkPlanGraph {
...
+    val sqlPlanPlaceHolder = "%s"
+    val htmlLineBreak = """<br align="left"/>""" + "\n"
+    val queryLabelFormat = 
+      s"""<<table border="0">
+         |<tr><td>Application: $appId, Query: $sqlId</td></tr>
+         |<tr><td>$sqlPlanPlaceHolder</td></tr>
+         |<tr><td href="$sqlPlanFile">
```
Comment (tgravescs) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateDot.scala:302]: I would just say see the output from --print-plans and not do the link.

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateDot.scala (hunk 1 above)
Comment (gerashegalov) [reply tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateDot.scala:295]: yes forgot about non-OS-mounted locations.

**Comment by gerashegalov** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**